testing/testnode.go

```diff
@@ -0,0 +1,94 @@
+package testing
+
+import (
+	"encoding/hex"
+	"net/http"
+	"path/filepath"
+	"time"
+
+	"go.sia.tech/renterd/internal/consensus"
+	"go.sia.tech/renterd/node"
+	"lukechampine.com/frand"
+)
+
+type noopHandler struct{}
+
+func (h *noopHandler) ServeHTTP(http.ResponseWriter, *http.Request) {
+}
+
+type TestCluster struct {
+	Autopilot *node.Node
+	Bus       *node.Node
+	Workers   []*node.Node
+
+	Cleanup []func() error
+}
+
+func randomPassword() string {
+	return hex.EncodeToString(frand.Bytes(32))
+}
+
+func newTestCluster(dir string) (*TestCluster, error) {
+	// Use shared wallet key.
+	wk := consensus.GeneratePrivateKey()
+
+	// Prepare individual dirs.
+	busDir := filepath.Join(dir, "bus")
+	workerDir := filepath.Join(dir, "worker")
+	autopilotDir := filepath.Join(dir, "autopilot")
+
+	// Generate API passwords.
+	busPassword := randomPassword()
+	workerPassword := randomPassword()
+	autopilotPassword := randomPassword()
+
+	// Create bus.
+	bus := node.NewNode("127.0.0.1:0", busPassword, busDir, &noopHandler{}, wk)
+	err := bus.CreateBus(true, "127.0.0.1:0")
+	if err != nil {
+		return nil, err
+	}
+
+	// Create worker.
+	worker := node.NewNode("127.0.0.1:0", workerPassword, workerDir, &noopHandler{}, wk)
+	err = worker.AddBus(bus.APIAddress(), busPassword)
+	if err != nil {
+		return nil, err
+	}
+	if err := worker.CreateWorker(); err != nil {
+		return nil, err
+	}
+
+	// Create autopilot.
+	autopilot := node.NewNode("127.0.0.1:0", autopilotPassword, autopilotDir, &noopHandler{}, wk)
+	err = autopilot.AddBus(bus.APIAddress(), busPassword)
+	if err != nil {
+		return nil, err
+	}
+	if err := autopilot.AddWorker(worker.APIAddress(), workerPassword); err != nil {
+		return nil, err
+	}
+	if err := autopilot.CreateAutopilot(3 * time.Second); err != nil {
+		return nil, err
+	}
+
+	return &TestCluster{
```

Comment:
I think most of my comments about the `Node` and `Cluster` are because I feel that nodes should be defined on their own, and the linking should happen when constructing a `Cluster`.

So instead of:
```go
ap := node.NewNode(...)
ap.AddBus(...)
ap.AddWorker(...)
...
cluster := TestCluster{ap, bus, worker}
```
it would be more like
```
b := node.NewBus(...)
a := node.NewAutopilot(b)
w := node.NewWorker(...)

cluster := TestCluster{a, b}
cluster.addWorker(w)
```

Reply:
The idea is that when a worker gets added the autopilot communicates where it can find the bus.